### PR TITLE
Everywhere: Add unused and old-style-cast warnings to AK, Kernel, and Libraries

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
this is work towards #5487

During whis I've found some other (related things) things:
* we were casting m_raw to u32 to just cast it back to u64
* somewhere we were using size_t as a ptr_t instead of FlatPtr
* why are we casting nulltptrs?
* why are we casting constants over two corners and not just saying the constant directly?
* someone else should take a look at IntrusiveList<T, member>::node_to_value its horrifying
* God help me are these a lot of casts

### CODE REVIEW

**Comment by alimpfard** [comment]
> Everywhere: change c-castings to corresponding c++ casting

Not that I'm saying we shouldn't do this, but is there any real benefit to this? it feels rather silly to go to all this trouble (and further trouble) just to appease a warning that doesn't really do much in a project of this scope?

I've always found `reinterpret_cast<T>(expr)` to be seriously chunky for no reason `(T)expr` communicates the same thing in a C++ project anyway?

**Comment by Domkeykong** [comment]
A C-style cast can perform any combination of the following casts:
- `const_cast`
- `static_cast`
- `reinterpret_cast`

By explicitly telling the compiler what we want a `static_cast` we are still `const` correct.
Doing a `reinterpret_cast` is still Undefined Behavior, but you already knew that ;)
Instead you should looking into doing a `std::bit_cast`

File: AK/IntrusiveList.h [hunk 1]
```diff
@@ -273,7 +273,11 @@ inline typename IntrusiveList<T, member>::Iterator IntrusiveList<T, member>::end
 template<class T, IntrusiveListNode T::*member>
 inline T* IntrusiveList<T, member>::node_to_value(IntrusiveListNode& node)
 {
-    return (T*)((char*)&node - ((char*)&(((T*)nullptr)->*member) - (char*)nullptr));
+// FIXME: WTF is even going on here, why are we casting nullptrs?
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+    return reinterpret_cast<T*>((char*)&node - ((char*)&(((T*)nullptr)->*member) - (char*)nullptr));
```
Comment (ADKaster) [inline AK/IntrusiveList.h:279]: This is a hand rolled `offsetof`. I believe we can use `__builtin_offsetof` instead, to be more standards compliant.

Comment (ADKaster) [reply AK/IntrusiveList.h:279]: Actually it's not that simple. I don't know how to implement this particular operation with the pointer-to-member using offsetof. But the comment should be more like `// arcane magic here, get pointer to T given the list node pointer-to-member in the CRTP'd type T.`

**Comment by Nicholas-Baron** [comment]
`-Wunused` is enabled project wide due to #6342. 

Maybe `-Wuseless-cast` could lighten the work by picking out redundant casts.

**Comment by Hendiadyoin1** [comment]
I think if I start work on thiis again I should drop the old-style-cast checks, because that's way to tedious and some casts are just weird...

**Comment by alimpfard** [comment]
If you're talking about the manual resolution of indirect class member pointers, I need to fix them for a little thing I'm working on.
Though it'll probably just be riddled with `reinterpret_cast` anyway.

**Comment by Hendiadyoin1** [comment]
In general, they are every where, and sometimes its logical what it is supposed to do and sometimes you are wondering, why we cast at that point (enums in switch statements casted to integers to handle a default case, was a case if I remember correctly)

And I started so far back, I would have to restart with that flag again...

**Pull request closed without merging**